src/AppInstallerCLICore/Command.h

```diff
@@ -16,21 +17,22 @@
 #include <type_traits>
 #include <vector>
 
-
 namespace AppInstaller::CLI
 {
     struct CommandException
     {
-        // The message should be a localized string, but the parameters are currently not localized.
-        // We 'convert' the param to a localization independent view here.
-        CommandException(Resource::LocString message, std::string_view param) : m_message(std::move(message)), m_param(param) {}
+        // The message should be a localized string.
+        // The parameters can be either localized or not.
+        // We 'convert' the param to a localization independent view here if needed.
+        CommandException(Resource::LocString message, Resource::LocString param) : m_message(std::move(message)), m_param(param) {}
+        CommandException(Resource::LocString message, std::string_view param) : m_message(std::move(message)), m_param(Utility::LocIndString{ param }) {}
 
         const Resource::LocString& Message() const { return m_message; }
-        const Utility::LocIndString Param() const { return m_param; }
+        const auto Param() const { return m_param; }
 
     private:
         Resource::LocString m_message;
-        Utility::LocIndString m_param;
+        std::variant<Resource::LocString, Utility::LocIndString> m_param;

```

Comment:
>std::variant<Resource::LocString, Utility::LocIndString> m_param; [](start = 8, length = 65)

I was just touching this and I didn't think to do it then, but now it comes to mind that LocString should just derive from LocIndString.

Reply:
Why did this become a shared_ptr?

---
In reply to: [602594039](https://github.com/microsoft/winget-cli/pull/810#discussion_r602594039) [](ancestors = 602594039)

Reply:
I wanted it to be able to hold a LocString or a LocIndString so I made it a pointer. I tried with a unique_ptr but `ParseArgumentsStateMachine::ThrowIfError()` needed to copy it to throw so I made it shared. I need to check if I can move it instead.

Reply:
Changed to use a plain LocIndString (not a pointer).